java/compaction/compaction-job-creation/src/test/java/sleeper/compaction/job/creation/CreateCompactionJobsIT.java

```diff
@@ -112,6 +123,35 @@ public void shouldCompactAllFilesInSinglePartition() throws Exception {
                 });
     }
 
+    @Test
+    public void shouldSendAssignJobIdRequestToSqs() throws Exception {
+        // Given
+        tableProperties.set(COMPACTION_JOB_ID_ASSIGNMENT_COMMIT_ASYNC, "true");
+        FileReferenceFactory fileReferenceFactory = FileReferenceFactory.from(stateStore);
+        FileReference fileReference1 = fileReferenceFactory.rootFile("file1", 200L);
+        FileReference fileReference2 = fileReferenceFactory.rootFile("file2", 200L);
+        FileReference fileReference3 = fileReferenceFactory.rootFile("file3", 200L);
+        FileReference fileReference4 = fileReferenceFactory.rootFile("file4", 200L);
+        List<FileReference> files = List.of(fileReference1, fileReference2, fileReference3, fileReference4);
+        stateStore.addFiles(files);
+
+        // When
+        jobCreator().createJobs(tableProperties);
+
+        // Then
+        assertThat(stateStore.getFileReferencesWithNoJobId())
+                .usingRecursiveFieldByFieldElementComparatorIgnoringFields("lastStateStoreUpdateTime")
+                .containsExactlyInAnyOrderElementsOf(files);
+        assertThat(receiveJobQueueMessage().getMessages())
+                .extracting(this::readJobMessage).singleElement().satisfies(job -> {
+                    assertThat(job.getInputFiles()).containsExactlyInAnyOrder("file1", "file2", "file3", "file4");
+                    assertThat(job.getPartitionId()).isEqualTo("root");
+                    assertThat(receiveAssignJobIdQueueMessage().getMessages().stream()
+                            .map(this::readAssignJobIdRequest))
+                            .containsExactly(requestToAssignFilesToJobs(List.of(job), tableProperties.get(TABLE_ID)));
+                });
```

Comment:
This is a nitpick now, but it's gone back to how it was when I was confused that it was receiving the assign job ID requests inside the outer assertion.

I think it'd be clearer if the call to receive the assign job requests happened outside the nested assertion. It might help if you make the test helper return a list of CompactionJobIdAssignmentCommitRequest, and add a version of the helper that reads the jobs queue that returns a list of CompactionJob.

It's up to you if you want to make this change, I'm fine with merging it.